rustls/src/crypto/ring/tls12.rs

```diff
@@ -162,9 +162,16 @@ impl Tls12AeadAlgorithm for GcmAlgorithm {
         write_iv: &[u8],
         explicit: &[u8],
     ) -> Result<ConnectionTrafficSecrets, UnsupportedOperationError> {
-        Ok(ConnectionTrafficSecrets::Aes128Gcm {
-            key,
-            iv: gcm_iv(write_iv, explicit),
+        Ok(match self.0.key_len() {
+            16 => ConnectionTrafficSecrets::Aes128Gcm {
+                key,
+                iv: gcm_iv(write_iv, explicit),
```

Comment:
Nit: would prefer to extract `let iv = gcm_iv(write_iv, explicit);`.

Reply:
Done.